src/Queue.php

```diff
@@ -10,51 +10,59 @@
 use Yiisoft\Yii\Queue\Enum\JobStatus;
 use Yiisoft\Yii\Queue\Exception\AdapterConfiguration\AdapterNotConfiguredException;
 use Yiisoft\Yii\Queue\Message\MessageInterface;
+use Yiisoft\Yii\Queue\Middleware\Push\AdapterPushHandler;
+use Yiisoft\Yii\Queue\Middleware\Push\MessageHandlerPushInterface;
+use Yiisoft\Yii\Queue\Middleware\Push\MiddlewarePushInterface;
+use Yiisoft\Yii\Queue\Middleware\Push\PushMiddlewareDispatcher;
+use Yiisoft\Yii\Queue\Middleware\Push\PushRequest;
 use Yiisoft\Yii\Queue\Worker\WorkerInterface;
 
 final class Queue implements QueueInterface
 {
-    protected WorkerInterface $worker;
-    protected LoopInterface $loop;
-    protected ?AdapterInterface $adapter;
-    private LoggerInterface $logger;
-    private string $channelName;
+    /**
+     * @var array|array[]|callable[]|MiddlewarePushInterface[]|string[]
+     */
+    private array $middlewareDefinitions;
+    private AdapterPushHandler $adapterPushHandler;
 
     public function __construct(
-        WorkerInterface $worker,
-        LoopInterface $loop,
-        LoggerInterface $logger,
-        ?AdapterInterface $adapter = null,
-        string $channelName = QueueFactoryInterface::DEFAULT_CHANNEL_NAME
+        private WorkerInterface $worker,
+        private LoopInterface $loop,
+        private LoggerInterface $logger,
+        private PushMiddlewareDispatcher $pushMiddlewareDispatcher,
+        private ?AdapterInterface $adapter = null,
+        private string $channelName = QueueFactoryInterface::DEFAULT_CHANNEL_NAME,
+        MiddlewarePushInterface|callable|array|string ...$middlewareDefinitions
```

Comment:
can't you do that?
```suggestion
        private MiddlewarePushInterface|callable|array|string ...$middlewareDefinitions
```

Reply:
No, property promotion is not available for variadics